stable/spinnaker/README.md

```diff
@@ -55,5 +55,38 @@ for Spinnaker. If you want to add arbitrary clusters need to do the following:
       contexts:
       # Names of contexts available in the uploaded kubeconfig
       - my-context
+      # This is the context from the list above that you would like
+      # to deploy Spinnaker itself to.
+      deploymentContext: my-context
     ```
 
+## Customizing your installation
+
+### Manual
+While the default installation is ready to handle your Kubernetes deployments, there are
+many different integrations that you can turn on with Spinnaker. In order to customize
+Spinnaker, you can use the [Halyard](https://www.spinnaker.io/reference/halyard/) command line `hal` 
+to edit the configuration and apply it to what has already been deployed.
+
+Halyard has an in-cluster daemon that stores your configuration. You can exec a shell in this pod to
+make and apply your changes. The Halyard daemon is configured with a persistent volume to ensure that
+your configuration data persists any node failures, reboots or upgrades.
+
+For example:
+
+```shell
+$ helm install -n cd stable/spinnaker
+$ kubectl exec -it cd-spinnaker-halyard-0 bash
+spinnaker@cd-spinnaker-halyard-0:/workdir$ hal version list
+```
+
+### Automated
+If you have known set of commands that you'd like to run after the base config steps or if
+you'd like to override some settings before the Spinnaker deployment is applied, you can enable
+the `halyard.additionalConfig.enabled` flag. You will need to create a config map that contains a key
+contianing the `hal` commands you'd like to run. You can set the key via the config map name via `halyard.additionalConfig.configMapName` and the key via `halyard.additionalConfig.configMapKey`. The `DAEMON_ENDPOINT` environment variable can be used in your custom commands to
+get a prepopulated URL that points to your Halyard daemon within the cluster. For example:
+
+```shell
```

Comment:
Should this shell command be present in a ConfigMap map that we manually create?
If so, could we also provide a sample `yaml` for this ConfigMap, to make things a little bit easier to understan?

I'm imagining that we would need to do something like:

`vi my-custom-hal-cmds-configmap.yaml`

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-custom-halyard-config
data:
  custom_commands.sh: |
    hal --daemon-endpoint $DAEMON_ENDPOINT config security authn oauth2 enable
```

`kubectl apply -f my-custom-hal-cmds-configmap.yaml`

And then run 
```bash
helm install stable/spinnaker 
    --set halyard.additionalConfig.configMapName=my-custom-halyard-config 
    --set halyard.additionalConfig.configMapKey=custom_commands.sh
```

Is that right?